src/pipeline/src/manager/table.rs

```diff
@@ -0,0 +1,450 @@
+                    datatype_extension: None,
+                },
+                ColumnDef {
+                    name: PIPELINE_TABLE_CREATED_AT_COLUMN_NAME.to_string(),
+                    data_type: ColumnDataType::TimestampNanosecond as i32,
+                    is_nullable: false,
+                    default_constraint: vec![],
+                    semantic_type: SemanticType::Timestamp as i32,
+                    comment: "".to_string(),
+                    datatype_extension: None,
+                },
+            ],
+        )
+    }
+
+    /// Build the column schemas for inserting a row into the pipeline table.
+    fn build_insert_column_schemas() -> Vec<PbColumnSchema> {
+        vec![
+            PbColumnSchema {
+                column_name: PIPELINE_TABLE_PIPELINE_NAME_COLUMN_NAME.to_string(),
+                datatype: ColumnDataType::String.into(),
+                semantic_type: SemanticType::Tag.into(),
+                ..Default::default()
+            },
+            PbColumnSchema {
+                column_name: PIPELINE_TABLE_PIPELINE_SCHEMA_COLUMN_NAME.to_string(),
+                datatype: ColumnDataType::String.into(),
+                semantic_type: SemanticType::Tag.into(),
+                ..Default::default()
+            },
+            PbColumnSchema {
+                column_name: PIPELINE_TABLE_PIPELINE_CONTENT_TYPE_COLUMN_NAME.to_string(),
+                datatype: ColumnDataType::String.into(),
+                semantic_type: SemanticType::Tag.into(),
+                ..Default::default()
+            },
+            PbColumnSchema {
+                column_name: PIPELINE_TABLE_PIPELINE_CONTENT_COLUMN_NAME.to_string(),
+                datatype: ColumnDataType::String.into(),
+                semantic_type: SemanticType::Field.into(),
+                ..Default::default()
+            },
+            PbColumnSchema {
+                column_name: PIPELINE_TABLE_CREATED_AT_COLUMN_NAME.to_string(),
+                datatype: ColumnDataType::TimestampNanosecond.into(),
+                semantic_type: SemanticType::Timestamp.into(),
+                ..Default::default()
+            },
+        ]
+    }
+
+    fn query_ctx(table_info: &TableInfo) -> QueryContextRef {
+        QueryContextBuilder::default()
+            .current_catalog(table_info.catalog_name.to_string())
+            .current_schema(table_info.schema_name.to_string())
+            .build()
+            .into()
+    }
+
+    /// Compile a pipeline from a string.
+    pub fn compile_pipeline(pipeline: &str) -> Result<Pipeline<GreptimeTransformer>> {
+        let yaml_content = Content::Yaml(pipeline.into());
+        parse::<GreptimeTransformer>(&yaml_content)
+            .map_err(|e| CompilePipelineSnafu { reason: e }.build())
+    }
+
+    fn generate_pipeline_cache_key(schema: &str, name: &str, version: Option<&str>) -> String {
+        format!("{}-{}-{}", schema, name, version.unwrap_or("latest"))
```

Comment:
Could we use the value of the timestamp to avoid formatting the timestamp to a string first?
```suggestion
    fn generate_pipeline_cache_key(schema: &str, name: &str, version: Option<&str>) -> String {
        match version {
            Some(v) => format!("{}-{}-{}", schema, name, v.value()),
            None => format!("{}-{}-latest", schema, name)
        }
```